### PR TITLE
Add sink_size parameter for mha_fwd and varlen_mha_fwd api

### PR DESCRIPTION
**Pull request by LJ-underdog**
## Motivation
Add sink_size parameter for mha_fwd&varlen_mha_fwd api, because ck fmha support attn_sink. ck pr:https://github.com/ROCm/composable_kernel/pull/3368

## Technical Details

1.  This PR adds support for the sink_size parameter to the MHA (Multi-Head Attention) forward APIs (mha_fwd and varlen_mha_fwd) to enable attention sink functionality, aligning with the CK FMHA implementation.

2. Key Changes:

- Added sink_size parameter to all MHA forward function signatures

- Updated mask construction to include sink size information

- Modified window_size tuples from 2-element to 3-element (adding sink_size as third element)

## Test Plan

Aiter currently has no tests, testing is on ck

## Test Result

Local mha op test is successful.
## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

**Comment by LJ-underdog** [comment]
ck pr:https://github.com/ROCm/composable_kernel/pull/3368

File: csrc/py_itfs_cu/asm_mha_fwd.cu [hunk 1]
```diff
@@ -133,6 +133,7 @@ mha_fwd_args get_asm_fmha_fwd_args(bool has_lse,
                         batch_stride_o,
                         mask.left,
                         mask.right,
+                        0,
```
Comment (valarLip) [inline csrc/py_itfs_cu/asm_mha_fwd.cu]: had some comments here

File: csrc/cpp_itfs/mha_fwd_generate.py [hunk 2]
```diff
@@ -161,8 +169,10 @@
                                      bias_type,
                                      has_lse,
                                      has_dropout,
+                                     false,
```
Comment (valarLip) [inline csrc/cpp_itfs/mha_fwd_generate.py]: had some comments here

File: aiter/ops/mha.py [hunk 3]
```diff
@@ -1824,7 +1833,7 @@ def flash_attn_func(
     dropout_p=0.0,
     softmax_scale=None,
     causal=False,
-    window_size=(-1, -1),  # -1 means infinite context window
+    window_size=(-1, -1, 0),  # -1 means infinite context window
```
Comment (valarLip) [inline aiter/ops/mha.py]: document the meaning of this 0

**Comment by valarLip** [comment]
looks we failed ci..
/sgl-workspace/aiter/csrc/include/mha_fwd.h:40:27: error: excess elements in struct initializer
   40 |                           has_sink},
      |                           ^~~~~~~~
/sgl-workspace/aiter/csrc/include/mha_fwd.h:70:35: error: excess elements in struct initializer
   70 |                                   has_sink} 
      |                                   ^~~~~~~~
/sgl-workspace/aiter/aiter/jit/build/mha_batch_prefill_bf16_nlogits_nbias_mask_nlse_ndropout/blob/mha_fwd.cpp:80:38: error: use of undeclared identifier 'has_sink'

**Review by valarLip — APPROVED** [review]
